planning/behavior_velocity_planner/include/scene_module/occlusion_spot/occlusion_spot_utils.hpp

```diff
@@ -118,53 +142,98 @@ struct PossibleCollisionInfo
   }
 };
 
+lanelet::ConstLanelet toPathLanelet(const PathWithLaneId & path);
+
+// Note : consider offset_from_start_to_ego and safety margin for collision here
+inline void handleCollisionOffset(
+  std::vector<PossibleCollisionInfo> & possible_collisions, double offset, double margin)
+{
+  for (auto & pc : possible_collisions) {
+    pc.arc_lane_dist_at_collision.length -= offset;
+    pc.arc_lane_dist_at_collision.length -= margin;
+  }
+}
+
+inline double offsetFromStartToEgo(
+  const PathWithLaneId & path, const Pose & ego_pose, const int closest_idx)
+{
+  double offset_from_ego_to_closest = 0;
+  for (int i = 0; i < closest_idx; i++) {
+    const auto & curr_p = path.points[i].point.pose.position;
+    const auto & next_p = path.points[i + 1].point.pose.position;
```

Comment:
Is is possible that closest_idx will be the final point of the path.points?

Reply:
Considering the unity with other parts, `.at(i)/.at(i+1)` is better here

Reply:
this is to find out ego closest point so if it's final point of path then ego goals

Reply:
fixed at c0d409a